src/libraries/System.Net.Sockets/tests/FunctionalTests/UnixDomainSocketTest.cs

```diff
@@ -586,6 +586,54 @@ public async Task ReceiveFrom_EndPoints_Correct(bool useAsync)
             }
         }
 
+        [Fact]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/52124", TestPlatforms.iOS | TestPlatforms.tvOS | TestPlatforms.MacCatalyst)]
+        public void UnixDomainSocket_Receive_GetsCanceledByDispose()
+        {
+            string path = GetRandomNonExistingFilePath();
+            var endPoint = new UnixDomainSocketEndPoint(path);
+
+            using (var server = new Socket(AddressFamily.Unix, SocketType.Stream, ProtocolType.Unspecified))
+            using (var client = new Socket(AddressFamily.Unix, SocketType.Stream, ProtocolType.Unspecified))
+            {
+                server.Bind(endPoint);
+                server.Listen(1);
+
+                client.Connect(endPoint);
+                int msDelay = 100;
+                bool readFailed = false;
+                byte[] buffer = new byte[100];
+                using (Socket accepted = server.Accept())
+                {
+                    while (msDelay < 10_000)
+                    {
+                        Task disposeTask = Task.Run(() =>
+                        {
+                            Thread.Sleep(msDelay);
+                            client.Dispose();
+                        });
+
+                        try
+                        {
+                            client.Receive(buffer);
```

Comment:
If there's a regression that causes `TryUnblockSocket` to break, the test will hang at `client.Receive` and never time out.

Reply:
We can do some things like:

https://github.com/dotnet/runtime/blob/5d0536142062ffe5d1222b02f4ef1943f878cb4a/src/libraries/System.Net.Sockets/tests/FunctionalTests/SendReceive/SendReceive.cs#L1079-L1082

Reply:
I was thinking about it. However, I think we would notice it and fix it. XUnit now spits out warnings about long running tasks.